### ISSUE TITLE
Adding a full example to the tests data directory

### ISSUE THREAD

**Issue by Zsailer** [nextstrain/augur#540]
Hi all, 

While trying to add tests to this repo, I've noticed that multiple PRs needed example data for testing. To generate such data, I ran through one of the examples [here](https://nextstrain.org/docs/tutorials/zika) and moved the output into the `tests/data` dir. This made it much easier to create tests. 

Now my question is, should we have a complete generic test dataset that all tests can use (preferably through a pytest fixture)? Right now, many PRs are creating individual test data for their tests to use. Ideally, though, all tests would use the same source data and access that data the same way. 

If this seems reasonable, I think the best way forward is to use one of the examples mentioned above (perhaps the [Zika dataset](https://nextstrain.org/docs/tutorials/zika)) into `tests/data` and remove some of the other outdated datasets in that directory. Thoughts?

**Comment by jameshadfield** [comment]
Hi @Zsailer -- have you seen the Snakemake runs in [`./test/builds`](https://github.com/nextstrain/augur/tree/master/tests/builds)? Specifically, I think the [zika fasta](https://github.com/nextstrain/augur/blob/master/tests/builds/zika/data/zika.fasta) within augur contains fewer sequences than the zika-tutorial but achieves the same end goal as the zika-tutorial

**Comment by jameshadfield** [comment]
> Now my question is, should we have a complete generic test dataset that all tests can use 

I agree with this, but I'd use the data that's currently being used by the Snakemake runs in `./test/builds` as a starting point. Those snakemake files can be updated to refer to the `tests/data` location. cc @huddlej 

> preferably through a pytest fixture

Quick heads up that in terms of "integration tests", or those which test the augur CLI itself, we're currently diescussing internally whether to go for the `cram` approach ([augur example here](https://github.com/nextstrain/augur/compare/test-with-cram#diff-fb675f7120b4b645ab65f1c3fa6f2146)) or a pytest approach ([augur example here](https://github.com/nextstrain/augur/pull/535)). Thoughts welcome as we'll try to make a decision shortly.

**Comment by Zsailer** [comment]
@jameshadfield I totally missed `./test/builds`—thank you for pointing this out! Super helpful!

I've used [pytest-datafiles](https://github.com/omarkohl/pytest-datafiles) in the past for handling data directories. It's pretty sweet. Files listed in `mark` are copied to a tmp directory and accessible through the `datafiles` fixture. This handles all the boilerplate code you'd need to write to make copies of the files and expose them through a custom fixture. 

An example snippet looks like:
 
```python
@pytest.mark.datafiles(
    'tests/builds/zika/data/zika.fasta
)
def test_example(datafiles):
    # Read in the zika fasta file
    zika_data_path = datafiles / 'zika.fasta'
    zika_data = zika_data_path.read_text()
    
    # Do something with the zika fasta data
    ... 


```

**Comment by Zsailer** [comment]
> go for the cram approach or a pytest approach 

For integration testing, it probably doesn't matter too much. Pytests definitely makes **unit testing** easier and cleaner, but I'm not sure it's really meant for testing CLI (though there is no reason you can't—e.g. your example works great). A pytest fixtures that emulate a CLI might get cryptic or messy as you try to support more complex entrypoints.

**Comment by genehack** [comment]
In the end, we have implemented cram tests.

I'm going to close this issue out.
